src/txBaseMsgProcessor.cpp

```diff
@@ -0,0 +1,403 @@
+#include <txBaseMsgProcessor.h>
+#include <main.h>
+#include <util.h>
+#include <consensus/validation.h>
+
+#include <set>
+
+TxBaseMsgProcessor::TxBaseMsg_DataToProcess::TxBaseMsg_DataToProcess():
+    txBaseHash(), sourceNodeId(-1),
+    pTxBase(nullptr), pSourceNode(nullptr),
+    txBaseProcessingState(MempoolReturnValue::NOT_PROCESSED_YET), txBaseValidationState() {};
+
+TxBaseMsgProcessor::TxBaseMsgProcessor()
+{
+    this->reset();
+}
+TxBaseMsgProcessor::~TxBaseMsgProcessor()
+{
+    this->reset();
+}
+
+void TxBaseMsgProcessor::reset()
+{
+    {
+        LOCK(cs_rejectedTxes);
+        recentRejects.reset(nullptr);
+        recentRejects.reset(new CRollingBloomFilter(120000, 0.000001));
+    }
+
+    {
+        LOCK(cs_orphanTxes);
+        mapOrphanTransactions.clear();
+        mapOrphanTransactionsByPrev.clear();
+    }
+    return;
+}
+
+void TxBaseMsgProcessor::addTxBaseMsgToProcess(const CTransactionBase& txBase, CNodeInterface* pfrom)
+{
+    CInv inv(MSG_TX, txBase.GetHash());
+    pfrom->AddInventoryKnown(inv);
+
+    pfrom->StopAskingFor(inv);
+    mapAlreadyAskedFor.erase(inv);
```

Comment:
In previous implementation `mapAlreadyAskedFor.erase(inv)` was executed with lock `LOCK(cs_main);`
Please, check the usage for possible conflicts.

Reply:
It's fine, I believe previous usage was an overkill. I also re-run some integration tests with the thread-sanitizer on and got no issues (at least with the specific executions I performed).